src/Domain/CampaignRepository.php

```diff
@@ -443,32 +444,126 @@ public function updateCampaignFromSFData(Campaign $campaign, array $campaignData
         );
     }
 
+    /**
+     * @param array<string, string> $jsonMatchOneConditions
+     * @param array<string, string> $jsonMatchInListConditions
+     */
+    private function filterForSearch(
+        QueryBuilder $qb,
```

Comment:
worth adding a dockblock to say this doesn't accept an arbitrary $qb, it has to be the one from the search function.